src/WebJobs.Extensions.DurableTask/Correlation/TraceContextBaseExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using DurableTask.Core;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Correlation
+{
+    /// <summary>
+    /// TraceContextBase extension methods.
+    /// </summary>
+    public static class TraceContextBaseExtensions
+    {
+        /// <summary>
+        /// Create RequestTelemetry from the TraceContext.
+        /// </summary>
+        /// <param name="context">TraceContext.</param>
+        /// <returns>RequestTelemetry.</returns>
+        public static RequestTelemetry CreateRequestTelemetry(this TraceContextBase context)
+        {
+            var telemetry = new RequestTelemetry { Name = context.OperationName };
+            telemetry.Duration = context.Duration;
+            telemetry.Timestamp = context.StartTime;
+            telemetry.Id = context.TelemetryId;
+            telemetry.Context.Operation.Id = context.TelemetryContextOperationId;
+            telemetry.Context.Operation.ParentId = context.TelemetryContextOperationParentId;
+
+            return telemetry;
+        }
+
+        /// <summary>
+        /// Create DependencyTelemetry from the Activity.
+        /// </summary>
+        /// <param name="context">TraceContext.</param>
+        /// <returns>DependencyTelemetry.</returns>
+        public static DependencyTelemetry CreateDependencyTelemetry(this TraceContextBase context)
+        {
+            var telemetry = new DependencyTelemetry { Name = context.OperationName };
+            telemetry.Start(); // TODO Check if it is necessary.
```

Comment:
In general, before we merge this all to dev, I would like to make sure we have issues tracking all of the TODOs so that they don't just stay there forever. Doesn't have to be addressed now.

Reply:
This TODO is not needed. Removed.